rust/feature-flags/src/flag_matching.rs

```diff
@@ -293,22 +506,61 @@ impl FeatureFlagMatcher {
 
         let props = match row {
             Some(row) => row.properties.0,
-            None => HashMap::new(),
+            None => HashMap::new(), // TODO handle empty result instead of allocating a new HashMap
         };
 
-        self.cached_properties = Some(props.clone());
+        self.cached_person_properties = Some(props.clone());
 
         Ok(props)
     }
 
-    // async fn get_group_properties_from_cache_or_db(
-    //     &self,
-    //     team_id: i32,
-    //     group_index: usize,
-    //     properties: &Vec<PropertyFilter>,
-    // ) -> HashMap<String, Value> {
-    //     todo!()
-    // }
+    async fn get_group_properties_from_in_memory_cache_or_db(
+        &mut self,
+        team_id: TeamId,
+        group_type_index: GroupTypeIndex,
+    ) -> Result<HashMap<String, Value>, FlagError> {
+        // TODO which of these properties do we need to cache?
+        if let Some(cached_group_props) = self.cached_group_properties.clone() {
+            // TODO: Maybe we don't want to copy around all user properties, this will by far be the largest chunk
+            // of data we're copying around. Can we work with references here?
+            // Worst case, just use a Rc, like we do with the group type mappings.
+            return Ok(cached_group_props);
+        }
+
+        if self.database_client.is_none() {
+            return Err(FlagError::DatabaseUnavailable);
+        }
```

Comment:
I used the same shape as what we did in `get_person_properties`, but I don't really get where _these_ cached props are coming from.  Yes, I set them if I have to do a fetch but, again, how do I know they persist?  Aren't these caches created fresh each time this request is called?

cc @neilkakkar  I'd love to know more about your original vision with caching here.

Reply:
think you can get rid of this, there's no real caching needed here. It's a relic from copying over stuff from django.

Reply:
(see above comment)